liquidator/liquidator.js

```diff
@@ -249,50 +269,129 @@ class Liquidator {
       );
 
       // Simple version of inventory management: simulate the transaction and assume that if it fails, the caller didn't have enough collateral.
-      let liquidationResult, gasEstimation;
-      try {
-        [liquidationResult, gasEstimation] = await Promise.all([
-          liquidation.call({ from: this.account }),
-          liquidation.estimateGas({ from: this.account })
+      const notEnoughCollateral = await liquidation
+        .call({ from: this.account })
+        .then(() => false)
+        .catch(() => true);
+
+      // Attempts to swap some capital for tokenCurrency
+      // Mutates the state of notEnoughCollateral in this logical scope
+      if (notEnoughCollateral) {
+        const tokenCurrency = await this.empClient.getTokenCurrency();
+
+        // reserveWeiNeeded is a reverse calculation to estimate how much capital we need to get the amount of `tokensToLiquidate`
+        // While oneGweiReturn is used as a reference point to determine slippage
+        const [reserveWeiNeeded, oneGweiReturn] = await Promise.all([
+          this.oneInch.getExpectedReturn({
```

Comment:
just so I understand: why is this `getExpectedReturn`'s `fromToken` the `tokenCurrency`, not the `reserveCurrency`? Are we not wanting to trade from `reserveCurrency->`tokenCurrency`.

Reply:
aha. I see prehaps just add a comment to explain this because it's a bit confusing.